src/net/net.c

```diff
@@ -3626,14 +3626,14 @@ char* NET_get_json_string(Net* net) {
     return NULL;
 
   // Max size
-  max_size = (2*NET_BUFFER_SIZE +
-	      BUS_BUFFER_SIZE*BUS_NUM_JSON_FIELDS*net->num_buses +
-	      BRANCH_BUFFER_SIZE*BRANCH_NUM_JSON_FIELDS*net->num_branches +
-	      GEN_BUFFER_SIZE*GEN_NUM_JSON_FIELDS*net->num_gens +
-	      LOAD_BUFFER_SIZE*LOAD_NUM_JSON_FIELDS*net->num_loads +
-	      SHUNT_BUFFER_SIZE*SHUNT_NUM_JSON_FIELDS*net->num_shunts +
-	      VARGEN_BUFFER_SIZE*VARGEN_NUM_JSON_FIELDS*net->num_vargens +
-	      BAT_BUFFER_SIZE*BAT_NUM_JSON_FIELDS*net->num_bats)*net->num_periods;
+  max_size = (2*NET_SINGLE_JSON_BUFFER_SIZE +
+	      (BUS_JSON_BUFFER_SIZE+JSON_STR_BUFFER_EXTRA)*BUS_NUM_JSON_FIELDS*net->num_buses +
+	      (BRANCH_JSON_BUFFER_SIZE+JSON_STR_BUFFER_EXTRA)*BRANCH_NUM_JSON_FIELDS*net->num_branches +
+	      (GEN_JSON_BUFFER_SIZE+JSON_STR_BUFFER_EXTRA)*GEN_NUM_JSON_FIELDS*net->num_gens +
+	      (LOAD_JSON_BUFFER_SIZE+JSON_STR_BUFFER_EXTRA)*LOAD_NUM_JSON_FIELDS*net->num_loads +
+	      (SHUNT_JSON_BUFFER_SIZE+JSON_STR_BUFFER_EXTRA)*SHUNT_NUM_JSON_FIELDS*net->num_shunts +
+	      (VARGEN_JSON_BUFFER_SIZE+JSON_STR_BUFFER_EXTRA)*VARGEN_NUM_JSON_FIELDS*net->num_vargens +
+	      (BAT_JSON_BUFFER_SIZE+JSON_STR_BUFFER_EXTRA)*BAT_NUM_JSON_FIELDS*net->num_bats)*net->num_periods;
```

Comment:
It is not enough to use the "JSON" buffer sizes? Do we still need to use the "EXTRA"? It would be great if we can remove the need use of the "EXTRA" everythere.

Also, where is the number 2 coming from? This was there from before. Maybe this has to do with the number of network fields that are not part of network components, i.e. base_power, version, and num_periods (see net.c). If so, then we should change it to 3